src/hotspot/os/windows/memMapPrinter_windows.cpp

```diff
@@ -0,0 +1,256 @@
+            buffer[IX] = 'x';
+            buffer[idx++] = 'c';
+        } else if (prot & PAGE_NOACCESS) {
+            strncpy_s(buffer, bufsiz, "(NA)", 5);
+        } else if (prot & PAGE_READONLY) {
+            buffer[IR] = 'r';
+        } else if (prot & PAGE_READWRITE) {
+            buffer[IR] = 'r';
+            buffer[IW] = 'w';
+        } else if (prot & PAGE_WRITECOPY) {
+            buffer[IW] = 'w';
+            buffer[idx++] = 'c';
+        } else if (prot & PAGE_TARGETS_INVALID) {
+            buffer[idx++] = 'i';
+        } else if (prot & PAGE_TARGETS_NO_UPDATE) {
+            buffer[idx++] = 'n';
+        } else if (prot != 0) {
+            snprintf(buffer, bufsiz, "(0x%x)", prot);
+            idx = static_cast<int>(strlen(buffer));
+        }
+
+        if (prot & PAGE_GUARD) {
+            buffer[idx++] = 'G';
+        }
+        if (prot & PAGE_NOCACHE) {
+            buffer[idx++] = 'C';
+        }
+        if (prot & PAGE_WRITECOMBINE) {
+            buffer[idx++] = 'W';
+        }
+        buffer[idx] = 0;
+        return buffer;
+    }
+    const char* getStateString(char* buffer, size_t bufsiz, MEMORY_BASIC_INFORMATION& mInfo) {
+        if (mInfo.State == MEM_COMMIT) {
+            buffer[0] = 'c'; buffer[1] = 0;
+        } else if (mInfo.State == MEM_FREE) {
+            buffer[0] = 'f'; buffer[1] = 0;
+        } else if (mInfo.State == MEM_RESERVE) {
+            buffer[0] = 'r'; buffer[1] = 0;
+        } else {
+            snprintf(buffer, bufsiz, "0x%x", mInfo.State);
+        }
+        return buffer;
+    }
+
+    const char* getTypeString(char* buffer, size_t bufsiz, MEMORY_BASIC_INFORMATION& mInfo) {
+        if (mInfo.Type == MEM_IMAGE) {
+            strncpy_s(buffer, bufsiz, "img", 4);
+        } else if (mInfo.Type == MEM_MAPPED) {
+            strncpy_s(buffer, bufsiz, "map", 4);
+        } else if (mInfo.Type == MEM_PRIVATE) {
+            strncpy_s(buffer, bufsiz, "pvt", 4);
+        } else {
+            snprintf(buffer, bufsiz, "0x%x", mInfo.Type);
+        }
+        return buffer;
+    }
+};
+
+class MappingInfoSummary {
+  unsigned _num_mappings;
+  size_t _total_region_size;  // combined resident set size
+  size_t _total_committed;    // combined committed size
+  size_t _total_reserved;     // combined shared size
+public:
+  MappingInfoSummary() : _num_mappings(0),  _total_region_size(0),
+                      _total_committed(0), _total_reserved(0) {}
+
+  void add_mapping(const MEMORY_BASIC_INFORMATION& mInfo, const MappingInfo& info) {
+    if (mInfo.State != MEM_FREE) {
+      _num_mappings++;
+      _total_region_size += mInfo.RegionSize;
+      _total_committed += mInfo.State == MEM_COMMIT ? mInfo.RegionSize : 0;
+      _total_reserved += mInfo.State == MEM_RESERVE ? mInfo.RegionSize : 0;
+    }
+  }
+
+  void print_on(const MappingPrintSession& session) const {
+    outputStream* st = session.out();
+    os::print_os_info(st);
+    os::print_memory_info(st);
```

Comment:
The "danger" here is that both of these functions may bloat in the future and print a lot more than today; and that we may want to use this printout in places where we also print out OS info and memory info, and therefore get duplicated printout (e.g. hs-err printing or jcmd VM.info). 

But I would print out the number of mappings you counted, and the total size of committed regions.

Reply:
I have rolled my own "print_os_info" from the utility functions, but have left print_memory_info in for now - if you want, I will remove it entirely, but I think the information provided is valuable to readers of this report.

Reply:
I've also printed the region count.